app/scripts/modules/core/utils/timeFormatters.ts

```diff
@@ -0,0 +1,84 @@
+import { IComponentController, IComponentOptions, module } from 'angular';
+import * as moment from 'moment';
+
+import { SETTINGS } from 'core/config/settings';
+
+export function duration(input: any) {
+  if (!input || isNaN(input) || input < 0) {
+    return '-';
+  }
+  const thisMoment = moment.utc(isNaN(parseInt(input, 10)) ? input : parseInt(input, 10));
```

Comment:
Do we need the `isNaN` check here? Seems like we'd short-circuit above so this extra check is unnecessary.